### PR TITLE
PLANNER-2596 CI jobs: consistent naming

### PR DESCRIPTION
**Pull request by Ginxo**
https://issues.redhat.com/browse/PLANNER-2596

* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1857
* https://github.com/kiegroup/kie-soup/pull/246
* https://github.com/kiegroup/droolsjbpm-knowledge/pull/586
* https://github.com/kiegroup/drools/pull/4087
* https://github.com/kiegroup/optaplanner/pull/1746
* https://github.com/kiegroup/lienzo-core/pull/161
* https://github.com/kiegroup/lienzo-tests/pull/134
* https://github.com/kiegroup/appformer/pull/1225
* https://github.com/kiegroup/kie-uberfire-extensions/pull/131
* https://github.com/kiegroup/jbpm/pull/2082
* https://github.com/kiegroup/droolsjbpm-integration/pull/2676
* https://github.com/kiegroup/kie-wb-playground/pull/127
* https://github.com/kiegroup/kie-wb-common/pull/3715
* https://github.com/kiegroup/drools-wb/pull/1531
* https://github.com/kiegroup/jbpm-designer/pull/926
* https://github.com/kiegroup/jbpm-work-items/pull/230
* https://github.com/kiegroup/jbpm-wb/pull/1532
* https://github.com/kiegroup/optaplanner-wb/pull/412
* https://github.com/kiegroup/kie-wb-distributions/pull/1151
* https://github.com/kiegroup/openshift-drools-hacep/pull/127
* https://github.com/kiegroup/process-migration-service/pull/41
* https://github.com/kiegroup/optaweb-employee-rostering/pull/745
* https://github.com/kiegroup/optaweb-vehicle-routing/pull/629
* https://github.com/kiegroup/kie-jpmml-integration/pull/73
* https://github.com/kiegroup/kie-docs/pull/4073

8.x:
* https://github.com/kiegroup/droolsjbpm-knowledge/pull/587
* https://github.com/kiegroup/drools/pull/4088
* https://github.com/kiegroup/optaplanner/pull/1747
* https://github.com/kiegroup/optaplanner/pull/1759
* https://github.com/kiegroup/optaweb-employee-rostering/pull/746
* https://github.com/kiegroup/optaweb-vehicle-routing/pull/630

### CODE REVIEW

**Review by triceo — APPROVED** [review]

File: .github/workflows/os_java_ci.yml [hunk 1]
```diff
@@ -20,19 +20,20 @@ defaults:
...
   os-java:
-    name: OS x Java
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest]
-        java-version: [8, 11]
-        maven-version: ['3.8.1']
+      os: [ubuntu-latest, windows-latest]
+      java-version: [8, 11]
+      maven-version: ['3.8.1']
```
Comment (rsynek) [inline .github/workflows/os_java_ci.yml:27]: The workflow run complains that the file is not valid.
```suggestion
        os: [ubuntu-latest, windows-latest]
        java-version: [8, 11]
        maven-version: ['3.8.1']
```

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
         maven-version: ['3.8.1']
       fail-fast: false
     runs-on: ${{ matrix.os }}
-    name: Maven Build
+    name: Maven - ${{ matrix.os }} - ${{ matrix.java-version }}
```
Comment (rsynek) [inline .github/workflows/pull_request.yml:26]: This job seems to have been already broken; see https://github.com/kiegroup/optaplanner/actions/runs/1626493390/workflow

**Review by rsynek — CHANGES_REQUESTED** [review]
The workflow

**Comment by Ginxo** [comment]
fixed by 
Screenshot from 2022-01-03 10-18-46

**Review by rsynek — APPROVED** [review]

**Comment by yurloc** [comment]
A couple of comments:
- "Maven" seems redundant. It doesn't add any extra information. It's _the build tool_ for the project and so it's in each combination.
- It's not exactly clear what 8 and 11 refer to. To avoid ambiguity, I would prefer "Java 11".

File: .github/workflows/jpa_integration_ci.yml [hunk 3]
```diff
@@ -35,16 +40,16 @@ jobs:
         ports:
           # Maps tcp port 5432 on service container to the host
           - 5432:5432
-
+    name: Maven - ${{ matrix.os }} - ${{ matrix.java-version }}
```
Comment (yurloc) [inline .github/workflows/jpa_integration_ci.yml:43]:
```suggestion
    name: ${{ matrix.os }} - Java ${{ matrix.java-version }}
```

File: .github/workflows/os_java_ci.yml [hunk 4]
```diff
@@ -20,19 +20,20 @@ defaults:
...
     runs-on: ${{ matrix.os }}
+    name: Maven - ${{ matrix.os }} - ${{ matrix.java-version }}
```
Comment (yurloc) [inline .github/workflows/os_java_ci.yml:29]:
```suggestion
    name: ${{ matrix.os }} - Java ${{ matrix.java-version }}
```

File: .github/workflows/pull_request.yml [hunk 5]
```diff
@@ -12,18 +12,18 @@ on:
...
+      cancel-in-progress: true
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
         java-version: [8, 11]
         maven-version: ['3.8.1']
       fail-fast: false
     runs-on: ${{ matrix.os }}
-    name: Maven Build
+    name: Maven - ${{ matrix.os }} - ${{ matrix.java-version }}
```
Comment (yurloc) [inline .github/workflows/pull_request.yml:26]:
```suggestion
    name: ${{ matrix.os }} - Java ${{ matrix.java-version }}
```

**Comment by Ginxo** [comment]
> A couple of comments:
> 
>     * "Maven" seems redundant. It doesn't add any extra information. It's _the build tool_ for the project and so it's in each combination.
> 
>     * It's not exactly clear what 8 and 11 refer to. To avoid ambiguity, I would prefer "Java 11".

I guess we should all agree on this and spread it to the rest of projects. @ge0ffrey @rsynek @mareknovotny wdyt?

**Comment by triceo** [comment]
Sure, why not.

**Comment by rsynek** [comment]
@Ginxo `Maven` is mostly redundant in the name, but in optaplanner-quickstarts, there are also `Gradle` builds.

+1 to use "Java ${version}"

**Comment by Ginxo** [comment]
@rsynek @triceo @yurloc what about these options, please choose one of them

1. `${{ matrix.os }} - Maven/Java ${{ matrix.java-version }}` or `${{ matrix.os }} - Gradle/Java ${{ matrix.java-version }}`
2. `${{ matrix.os }} - Java ${{ matrix.java-version }}`
3. `Maven/Java ${{ matrix.java-version }} - ${{ matrix.os }}`
4. `Java ${{ matrix.java-version }} - ${{ matrix.os }}`

it is important to reach a consensus before spread it out for dozens of PRs, so please vote, thanks!

**Comment by triceo** [comment]
I am fine with either, as long as it is consistent. If I absolutely have to pick, I pick 3.

**Comment by rsynek** [comment]
The most important question for me is whether we could and should omit the `Maven` in the name. If we do, `Gradle` becomes an exception to the rule, which I would rather avoid.
How about making the build tool the least significant part of the name by moving it to the end?
`${{ matrix.os }} - Java ${{ matrix.java-version }} / (Maven|Gradle)`

**Comment by yurloc** [comment]
> The most important question for me is whether we could and should omit the `Maven` in the name. If we do, `Gradle` becomes an exception to the rule, which I would rather avoid. How about making the build tool the least significant part of the name by moving it to the end? `${{ matrix.os }} - Java ${{ matrix.java-version }} / (Maven|Gradle)`

OK, but let's **consistently only use `-` as a delimiter**. Perfect is the enemy of good -- we can change the order of components later. Other than that Radek's proposal LGTM.

The corrected version with `-` delimiter:
`${{ matrix.os }} - Java ${{ matrix.java-version }} - (Maven|Gradle)`

**Comment by triceo** [comment]
We're bikeshedding. Pick any one and apply it consistently, I'll be happy.

**Comment by Ginxo** [comment]
done for all. I've even created a new PR for optaplanner main branch https://github.com/kiegroup/optaplanner/pull/1759 (since it was already merged)

**Review by triceo — APPROVED** [review]

**Review by yurloc — APPROVED** [review]

**Pull request merged**